src/inspect_ai/hooks/_hooks.py

```diff
@@ -271,6 +271,13 @@ async def on_model_usage(self, data: ModelUsageData) -> None:
         """
         pass
 
+    async def on_sample_score(self) -> None:
```

Comment:
We need a data element passed here that has the tree of ids that uniquely identify the scoring context (for hooks that are stateless and write through to a database). So something like this:

```python
@dataclass(frozen=True)
class SampleScore:
    """Sample score hook event data."""
    eval_set_id: str | None
    """The globally unique identifier for the eval set (if any)."""
    run_id: str
    """The globally unique identifier for the run."""
    eval_id: str
    """The globally unique identifier for the task execution."""
    sample_id: str
    """The globally unique identifier for the sample execution."""
```

Reply:
Yes your custom hook using the `store()` is completely fine, but we don't want to presume that approach (i.e. a lot of hooks may be writing straight through to a database or log file).

Reply:
Okay, sure, will add that functionality!